src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -161,6 +161,7 @@ internal interface ISyntaxFacts
         /// </summary>
         bool IsPreprocessorKeyword(SyntaxToken token);
         bool IsPreProcessorDirectiveContext(SyntaxTree syntaxTree, int position, CancellationToken cancellationToken);
+        bool IsIdentifierContainerPreprocessorDirectiveTrivia(SyntaxNode node);
```

Comment:
not a syntax fact.

Reply:
Where would it be placed then? Because it is a crucial part of determining whether to look for preprocessing symbols or not.

Reply:
> Because it is a crucial part of determining

Why is it crucial?  Who/where needs this info?

Reply:
The `GetDeclaredSymbol` extension uses this, terminating early and constructing a preprocessing symbol for the given identifier.

Reply:
> Why is it crucial? Who/where needs this info?